### PR TITLE
[core] fix collisionBox alignment when Icon/text translation is enabled

### PR DESCRIPTION
**Pull request by zmiao**
Fix partially of  #13526 
Render test: https://github.com/mapbox/mapbox-gl-js/pull/8659

### CODE REVIEW

File: src/mbgl/layout/symbol_layout.cpp [hunk 1]
```diff
@@ -850,27 +856,29 @@ void SymbolLayout::addToDebugBuffers(SymbolBucket& bucket) {
...
+                    collisionBox->lines.emplace_back(index + 2, index + 3);
+                    collisionBox->lines.emplace_back(index + 3, index + 0);
                 }
 
                 segment.vertexLength += vertexLength;
                 segment.indexLength += indexLength;
             }
         };
-        populateCollisionBox(symbolInstance.textCollisionFeature);
+        populateCollisionBox(symbolInstance.textCollisionFeature, true);
```
Comment (pozdnyakov) [inline src/mbgl/layout/symbol_layout.cpp]: nit: `true /*isText*/`

File: src/mbgl/layout/symbol_layout.cpp [hunk 2]
```diff
@@ -850,27 +856,29 @@ void SymbolLayout::addToDebugBuffers(SymbolBucket& bucket) {
...
         if (symbolInstance.verticalTextCollisionFeature) {
-            populateCollisionBox(*symbolInstance.verticalTextCollisionFeature);
+            populateCollisionBox(*symbolInstance.verticalTextCollisionFeature, true);
```
Comment (pozdnyakov) [inline src/mbgl/layout/symbol_layout.cpp]: ditto

File: src/mbgl/layout/symbol_layout.cpp [hunk 3]
```diff
@@ -850,27 +856,29 @@ void SymbolLayout::addToDebugBuffers(SymbolBucket& bucket) {
...
         }
         if (symbolInstance.verticalIconCollisionFeature) {
-            populateCollisionBox(*symbolInstance.verticalIconCollisionFeature);
+            populateCollisionBox(*symbolInstance.verticalIconCollisionFeature, false);
```
Comment (pozdnyakov) [inline src/mbgl/layout/symbol_layout.cpp]: ditto

File: src/mbgl/layout/symbol_layout.cpp [hunk 4]
```diff
@@ -850,27 +856,29 @@ void SymbolLayout::addToDebugBuffers(SymbolBucket& bucket) {
...
         }
-        populateCollisionBox(symbolInstance.iconCollisionFeature);
+        populateCollisionBox(symbolInstance.iconCollisionFeature, false);
```
Comment (pozdnyakov) [inline src/mbgl/layout/symbol_layout.cpp]: ditto

**Comment by astojilj** [comment]
> Looks nice! My only concern is memory usage, @zmiao do you have some estimations of the memory consumption impact caused by the change?

@pozdnyakov , @zmiao 
I didn't review the patch yet... what could cause increased memory consumption?

**Comment by pozdnyakov** [comment]
> what could cause increased memory consumption?

The [icon collision data](https://github.com/mapbox/mapbox-gl-native/pull/15467/files#diff-dbe8d2b21eecb65ce1f99ee0c9ddc4f4R143) might require more memory

**Comment by zmiao** [comment]
@pozdnyakov @astojilj the number of the vertexes remains the same, only change is the container is split as two which hold text and icon separately. So no extra memory is added.

**Comment by pozdnyakov** [comment]
> @pozdnyakov @astojilj the number of the vertexes remains the same, only change is the container is split as two which hold text and icon separately. So no extra memory is added.

I guess it's just `+ sizeof(CollisionBoxBuffer)` per bucket, think its fine. Thanks!

**Pull request merged**